examples/DGmethods/ex_001_dycoms.jl

```diff
@@ -0,0 +1,259 @@
+
+  param = init_ode_param(dg)
+
+  Q = init_ode_state(dg, param, DF(0))
+
+  lsrk = LSRK54CarpenterKennedy(dg, Q; dt = dt, t0 = 0)
+
+  eng0 = norm(Q)
+  @info @sprintf """Starting
+  norm(Q₀) = %.16e""" eng0
+
+  # Set up the information callback
+  starttime = Ref(now())
+  cbinfo = GenericCallbacks.EveryXWallTimeSeconds(60, mpicomm) do (s=false)
+    if s
+      starttime[] = now()
+    else
+      energy = norm(Q)
+      @info @sprintf("""Update
+                     simtime = %.16e
+                     runtime = %s
+                     norm(Q) = %.16e""", ODESolvers.gettime(lsrk),
+                     Dates.format(convert(Dates.DateTime,
+                                          Dates.now()-starttime[]),
+                                  Dates.dateformat"HH:MM:SS"),
+                     energy)
+    end
+  end
+
+  step = [0]
+    cbvtk = GenericCallbacks.EveryXSimulationSteps(1000) do (init=false)
+    mkpath("./vtk/")
+    outprefix = @sprintf("./vtk-dycoms/dycoms_%dD_mpirank%04d_step%04d", dim,
+                           MPI.Comm_rank(mpicomm), step[1])
+    @debug "doing VTK output" outprefix
+    writevtk(outprefix, Q, dg)
+        
+    step[1] += 1
+    nothing
+  end
+
+  solve!(Q, lsrk, param; timeend=timeend, callbacks=(cbinfo, cbvtk))
+
+  # Print some end of the simulation information
+  engf = norm(Q)
+  Qe = init_ode_state(dg, param, DF(timeend))
+
+  engfe = norm(Qe)
+  errf = euclidean_distance(Q, Qe)
+  @info @sprintf """Finished
+  norm(Q)                 = %.16e
+  norm(Q) / norm(Q₀)      = %.16e
+  norm(Q) - norm(Q₀)      = %.16e
+  norm(Q - Qe)            = %.16e
+  norm(Q - Qe) / norm(Qe) = %.16e
+  """ engf engf/eng0 engf-eng0 errf errf / engfe
+  errf
+end
+
+using Test
+let
+  MPI.Initialized() || MPI.Init()
+  mpicomm = MPI.COMM_WORLD
+  ll = uppercase(get(ENV, "JULIA_LOG_LEVEL", "INFO"))
+  loglevel = ll == "DEBUG" ? Logging.Debug :
+    ll == "WARN"  ? Logging.Warn  :
+    ll == "ERROR" ? Logging.Error : Logging.Info
+  logger_stream = MPI.Comm_rank(mpicomm) == 0 ? stderr : devnull
+  global_logger(ConsoleLogger(logger_stream, loglevel))
+  @static if haspkg("CUDAnative")
+      device!(MPI.Comm_rank(mpicomm) % length(devices()))
+  end
+  
+  # Problem type
+  DF = Float64
+  # DG polynomial order 
+  polynomialorder = 4
+  # User specified grid spacing
+  Δx    = DF(35)
+  Δy    = DF(35)
+  Δz    = DF(10)
+  # SGS Filter constants
+  C_smag = DF(0.15)
+  Δ     = DF(cbrt(Δx * Δy * Δz))
+  # Physical domain extents 
+  (xmin, xmax) = (0, 2000)
+  (ymin, ymax) = (0, 2000)
+  (zmin, zmax) = (0, 1500)
+  #Get Nex, Ney from resolution
+  Lx = xmax - xmin
+  Ly = ymax - ymin
+  Lz = zmax - ymin
+  # User defines the grid size:
+  ratiox = (Lx/Δx - 1)/polynomialorder
+  ratioy = (Ly/Δy - 1)/polynomialorder
+  ratioz = (Lz/Δz - 1)/polynomialorder
+  Nex = ceil(Int64, ratiox)
+  Ney = ceil(Int64, ratioy)
+  Nez = ceil(Int64, ratioz)
+  Ne = (Nex, Ney, Nez)
+  # User defined domain parameters
```

Comment:
It would be nice if the `Lx`, `ratiox`, `Nex` etc. do not need to be defined here but are calculated in some lower layer from `Delta x` etc.